.github/workflows/build.yaml

```
@@ -30,3 +30,9 @@ jobs:
           fetch-depth: 1
       - name: Build
         run: make
+      - name: Run markdownlint with auto-fix
+        uses: DavidAnson/markdownlint-cli2-action@v19
```

As an OSSF best practice we prefer to specify actions by SHA and not by version. You can look up the version like this:

```console
❯ gh api repos/DavidAnson/markdownlint-cli2-action/git/ref/tags/v19 | jq -r .object.sha
05f32210e84442804257b2a6f20b273450ec8265
```

Dependebot will take care of updating from then on.

```suggestion
        uses: DavidAnson/markdownlint-cli2-action@05f32210e84442804257b2a6f20b273450ec8265 # v19
```

---

Thanks , I'll fix it

---

@jandubois using digest looks good (I did not know that dependebot supports this) but do you have any references to this best practice?

---

You can find it under [pinned dependencies](https://github.com/ossf/scorecard/blob/main/docs/checks.md#pinned-dependencies):

> For GitHub workflows used in building and releasing your project, pin dependencies by hash. See [main.yaml](https://github.com/ossf/scorecard/blob/f55b86d6627cc3717e3a0395e03305e81b9a09be/.github/workflows/main.yml#L27) for example. To determine the permissions needed for your workflows, you may use [StepSecurity's online tool](https://app.stepsecurity.io/secureworkflow/) by ticking the "Pin actions to a full length commit SHA". You may also tick the "Restrict permissions for GITHUB_TOKEN" to fix issues found by the Token-Permissions check.

---

> Dependebot will take care of updating from then on.

I just noticed that we have not set up dependabot for this repo yet; will create an issue for it.

---

I have created a pullrequest for it as well :) and fixed this one too